app/src/test/java/tech/ula/utils/ValidationUtilityTest.kt

```diff
@@ -0,0 +1,229 @@
+package tech.ula.utils
+
+import android.content.res.Resources
+import com.nhaarman.mockitokotlin2.whenever
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import tech.ula.R
+
+@RunWith(MockitoJUnitRunner::class)
+class ValidationUtilityTest {
+
+    @Mock
+    lateinit var validationUtility: ValidationUtility
+
+    @Mock
+    lateinit var credential: Credential
+
+    @Mock
+    lateinit var resources: Resources
+
+    private val defaultPassword = "password"
+
+    @Before
+    fun setup() {
+        validationUtility = ValidationUtility(resources)
+
+        val blacklistedUsernames = arrayOf(
+                "adm",
+                "audio",
+                "autologin",
+                "backup",
+                "bin",
+                "cdrom",
+                "daemon",
+                "dbus",
+                "dialout",
+                "dip",
+                "disk",
+                "fax",
+                "floppy",
+                "ftp",
+                "games",
+                "gnats",
+                "http",
+                "input",
+                "irc",
+                "kmem",
+                "kvm",
+                "list",
+                "locate",
+                "lock",
+                "log",
+                "lp",
+                "lpadmin",
+                "mail",
+                "mem",
+                "netdev",
+                "network",
+                "news",
+                "nobody",
+                "nogroup",
+                "operator",
+                "optical",
+                "plugdev",
+                "postgres",
+                "power",
+                "proc",
+                "proxy",
+                "rfkill",
+                "root",
+                "sambashare",
+                "sasl",
+                "scanner",
+                "shadow",
+                "smmsp",
+                "src",
+                "staff",
+                "storage",
+                "sudo",
+                "sync",
+                "sys",
+                "systemd-journal",
+                "tape",
+                "tty",
+                "users",
+                "utmp",
+                "uucp",
+                "uuidd",
+                "vboxusers",
+                "video",
+                "voice",
+                "wheel",
+                "www-data"
+        )
+
+        whenever(resources.getStringArray(R.array.blacklisted_usernames))
```

Comment:
You stub the call for `R.array.blacklisted_usernames` twice. I admire the intent to have the whole list here, but again, we're testing _behavior_ and not necessarily exact implementation. We can assume in a real scenario that a call to `resources.getString(R.array.blacklisted_usernames)` will successfully fetch the array, and therefore only need to test it with an array of size 1. This also saves us from needing to keep the lists in sync.

Reply:
That makes sense, I've changed it to be an array of one.  `root`